app/jobs/searchgov_domain_destroyer_job.rb

```diff
@@ -1,11 +1,34 @@
 class SearchgovDomainDestroyerJob < ApplicationJob
   queue_as :searchgov
 
-  def perform(searchgov_domain:)
-    searchgov_domain.searchgov_urls.find_each do |url|
-      url.destroy!
+  def perform(searchgov_domain)
+    ActiveRecord::Base.transaction do
+      if destroy_associated_records(searchgov_domain.searchgov_urls, 'URL') && destroy_record(searchgov_domain, 'SearchgovDomain')
+        Rails.logger.info("Successfully destroyed SearchgovDomain #{searchgov_domain.id} and its URLs.")
+      else
+        Rails.logger.error("Failed to completely destroy SearchgovDomain #{searchgov_domain.id} and its URLs.")
+      end
     end
-    searchgov_domain.destroy!
   end
-end
 
+  private
+
+  def destroy_associated_records(records, record_type)
+    records.find_each(batch_size: 100) { |record| return false unless destroy_record(record, record_type) }
+
+    true
+  end
+
+  def destroy_record(record, record_type)
+    unless record.destroy
```

Comment:
@stevenbarragan  - I see your point, however, the current approach has a specific rationale behind it.

Rails calls destroy on an ActiveRecord object will attempt to destroy the object and its associated dependent records, this will handle hooks (like before_destroy or after_destroy) that throw an :abort symbol, then the destruction process will be halted, no exceptions are raised, and destroy will return false.

In the current implementation, checking the return value of destroy allows it to gracefully handle situations where the record could not be destroyed due to one of these hooks. If the destroy operation were to fail, it would catch it and log an appropriate error message, which is preferable over having an unhandled exception.